packages/cli-repl/src/cli-repl.ts

```diff
@@ -44,17 +44,37 @@ class CliRepl {
   private mongoshDir: string;
 
   /**
-   * Connect to the cluster.
+   * Instantiate the new CLI Repl.
+   */
+  constructor(driverUri: string, driverOptions: NodeOptions, options: CliOptions) {
+    this.options = options;
+    this.mongoshDir = path.join(os.homedir(), '.mongodb/mongosh/');
+
+    this.createMongoshDir();
+
+    this.bus = new Nanobus('mongosh');
+    logger(this.bus, this.mongoshDir);
+
+    this.generateOrReadTelemetryConfig();
+
+    if (this.isPasswordMissing(driverOptions)) {
+      this.requirePassword(driverUri, driverOptions);
+    } else {
+      this.setupRepl(driverUri, driverOptions);
+    }
+  }
+
+  /**
+   * setup CLI environment: serviceProvider, ShellEvaluator, log connection
+   * information, and finally start the repl.
    *
    * @param {string} driverUrl - The driver URI.
    * @param {NodeOptions} driverOptions - The driver options.
    */
-  async connect(driverUri: string, driverOptions: NodeOptions): Promise<void> {
-    console.log(i18n.__(CONNECTING), clr(redactPwd(driverUri), ['bold', 'green']));
-
-    this.serviceProvider = await CliServiceProvider.connect(driverUri, driverOptions);
+  async setupRepl(driverUri: string, driverOptions: NodeOptions): Promise<void> {
+    this.serviceProvider = await this.connect(driverUri, driverOptions);
     this.ShellEvaluator = new ShellEvaluator(this.serviceProvider, this.bus, this);
-    this.buildInfo = await this.serviceProvider.buildInfo();
+    this.buildInfo = await this.getBuild();
```

Comment:
why not `getBuildInfo`?

Reply:
i thought it'd be confusing, because there is another function (not on `this` scope though) that's already called `getBuildInfo`.

Although, I could refactor that other function to be `formatBuildInfo`, since it plucks out and returns only the necessary information from `buildInfo` and `getCmdLineOpts` functions.

Reply:
oh nvm, i thought there was a function, but there isn't one. I'll rename this to `getBuildInfo`